babel.config.js

```diff
@@ -1,53 +1,63 @@
+/**
+ * Babel
+ *
+ * @see https://gist.github.com/ncochard/6cce17272a069fdb4ac92569d85508f4
+ */
+
 module.exports = {
-  presets: ['@babel/react'],
+  presets: ['@babel/preset-env', '@babel/preset-react'],
   plugins: [
     '@babel/plugin-transform-runtime',
     '@babel/plugin-proposal-class-properties',
-    [
-      'styled-components',
-      {
-        displayName: true,
-        ssr: true
-      }
-    ],
-    [
-      'module-resolver',
-      {
-        root: ['./'],
-        alias: {
-          api: './api',
-          config: './config',
-          decorators: './.storybook/decorators',
-          root: './',
-          storybook: './.storybook',
-          themes: './.storybook/themes'
-        }
-      }
-    ]
+    'babel-plugin-add-react-displayname'
   ],
   env: {
-    build: {
+    test: {
+      presets: [
+        [
+          '@babel/preset-env'
+          // {
+          //   targets: {
+          //     node: 'current'
+          //   }
+          // }
+        ],
+        '@babel/preset-react'
+      ],
+      plugins: ['require-context-hook']
+    },
+    esm: {
       presets: [
-        'next/babel',
         [
-          '@babel/env',
+          '@babel/preset-env',
           {
             modules: false
+            // targets: {
+            //   node: 'current'
+            // }
           }
         ]
-      ]
-    },
-    test: {
-      presets: [
+      ],
+      plugins: [
         [
-          '@babel/env',
+          '@babel/plugin-transform-runtime',
           {
-            targets: {
-              node: 'current'
-            }
+            useESModules: true
           }
         ]
-      ]
+        // [
+        //   'styled-components',
+        //   {
+        //     // displayName: true,
+        //     // minify: true,
+        //     // namespace: 'cleverly',
+        //     // pure: true,
+        //     ssr: true
+        //     // transpileTemplateLiterals: true
+        //   }
+        // ]
```

Comment:
Let's remove commented codes to keep this as clean as possible.

Reply:
still testing this so want to keep the comments in

Reply:
gotcha